src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,79 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.InvalidDataException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    public User getDefaultValidUser() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setId(38628L);
+        user.setPassword("unuwjf8j2");
+        user.setAge(22);
+        return user;
+    }
+
+    @Test
+    void registrationDataIsValid() throws InvalidDataException {
+        RegistrationService registrationService = new RegistrationServiceImpl();
+        User user = getDefaultValidUser();
+        try {
+            registrationService.register(user);
+        } catch (InvalidDataException e) {
+            throw new InvalidDataException("Oops, your input data is not valid");
+        }
+    }
+
+    @Test
+    void ageIsNotOk() {
+        RegistrationService registrationService = new RegistrationServiceImpl();
+        User user = getDefaultValidUser();
+        user.setAge(16);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void passwordIsNotValid() {
+        RegistrationService registrationService = new RegistrationServiceImpl();
+        User user = getDefaultValidUser();
+        user.setPassword("321");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void loginIsNotValid() {
+        RegistrationService registrationService = new RegistrationServiceImpl();
+        User user = getDefaultValidUser();
+        user.setLogin("jfi");
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void inputDataIsNull() {
+        RegistrationService registrationService = new RegistrationServiceImpl();
+        User user = getDefaultValidUser();
+        user.setId(null);
+        user.setLogin(null);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
+
+    @Test
+    void userAlreadyExist() {
+        RegistrationService registrationService = new RegistrationServiceImpl();
+        User user = getDefaultValidUser();
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user);
+        });
+    }
```

Comment:
* The logic that you have here right now throws an exception only because your tests are not isolated enough and the users from other tests are staying in the `storage` after their completion. you need to clear it after/before each test.
* after you are done with the first point: You do not have a logic for putting a user that should already be present in the storage when you call the `register()` method. Hint: the first user should be put **not** using a `register()` but something more low-level because register is not guaranteed to work properly since it is under a test, check common mistakes for details on that